libs/platform/src/lib/components/form/text-area/text-area.component.ts

```diff
@@ -0,0 +1,395 @@
+
+    /** handle exceeded maxlength case when text is pasted */
+    handlePasteInteraction(): void {
+        this._isPasted = true;
+        /// For IE
+        if (window['clipboardData']) {
+            const value = window['clipboardData'].getData('Text');
+            // todo: handle for IE
+            this.updateCounterInteractions();
+        } else {
+            // for other navigators
+            navigator['clipboard'].readText().then((clipText) => {
+                this.updateCounterInteractions();
+            });
+        }
+    }
+
+    /** handle auto growing of textarea */
+    autoGrowTextArea() {
+        if (this.growing) {
+            this.textareaElement.nativeElement.style.height = 'inherit';
+            // console.log('maxheight' + this.textareaElement.nativeElement.style.maxHeight);
+            // console.log('scroll' + this.textareaElement.nativeElement.scrollHeight);
+            const textareaTotalHeight = this._getTextareaTotalHeight();
+            const maxHeight = parseInt(this.textareaElement.nativeElement.style.maxHeight as string, 10);
+            if (maxHeight) {
+                // console.log('total height' + textareaTotalHeight);
+                if (textareaTotalHeight <= maxHeight) {
+                    this.textareaElement.nativeElement.style.height = textareaTotalHeight + 'px';
+                } else {
+                    if (this.height) {
+                        this.textareaElement.nativeElement.style.height = this.height;
+                    } else {
+                        this.textareaElement.nativeElement.style.height = maxHeight + 'px';
+                    }
+                }
+            } else {
+                // no bound max height, keep growing
+                this.textareaElement.nativeElement.style.height = textareaTotalHeight + 'px';
+            }
+        }
+    }
+
+    /**
+     * get line height of textarea
+     */
+    getTextareaLineHeight(): number {
+        // Get the computed styles for the element
+        if (this.textareaElement && this.textareaElement.nativeElement) {
+            const computed = window.getComputedStyle(this.textareaElement.nativeElement);
+
+            // Calculate the height
+            return parseInt(computed.getPropertyValue('line-height'), 10);
+        }
+        return 20; // default line height if nothing is defined
+    }
+
+    /** when backpress or delete key is pressed when showExceededText field is not set, simply remove excess characters,
+    /* else handle autogrow case
+    */
+    @HostListener('keyup', ['$event'])
+    handleBackPress(event: KeyboardEvent) {
+        // if not showing exceeded text message/interactions or growing, and custom value set
+        if (this.growing) {
+            this.autoGrowTextArea();
+            this.isValueCustomSet = false; // set it to false first time it comes here
+        }
+        if (!this.showExceededText && (event.keyCode === DELETE || event.keyCode === BACKSPACE)) {
```

Comment:
Use key utils. We agreed that we will try to use keyUtils as much as we can and stop using keyCOde which is deprecated. Please see select for the usage (select-commons.ts)

Reply:
using KeyUtils now